Add Lineage and Motorola settings database To SetEdit.

Tharow opened the pull request:
Open to reformating code to match existing code as desired.

This could also be completed with by adding a settingsLocation along with settingstype in the SettingsRecyclerAdapter.
This alternate solution could be made if desired.

https://github.com/MuntashirAkon/SetEdit/issues/41

Tharow committed:
Added Lineage and motorola settings database sections
TODO: hide Unavailable Settings Databases
Manifest Lists Used queries

Tharow committed:
Null Fix
Permission is required next

Tharow committed:
Permissions Fix, And Command Fix

MuntashirAkon commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,18 +10,27 @@
     <uses-permission android:name="com.android.launcher.permission.INSTALL_SHORTCUT" />
 
+
+
```
Please remove these extra spaces.

MuntashirAkon commented on app/src/main/java/io/github/muntashirakon/setedit/EditorActivity.java:
```diff
@@ -155,7 +155,7 @@ public void onCreate(Bundle bundle) {
                     addNewItemDialog();
                 } else {
-                    EditorUtils.displayGrantPermissionMessage(this);
+                    EditorUtils.displayGrantPermissionMessage(this, ((SettingsRecyclerAdapter) adapter).getSettingsType());
```
You may want to store `((SettingsRecyclerAdapter) adapter).getSettingsType()` to another variable since there are more than one access in this scope.

MuntashirAkon commented on app/src/main/java/io/github/muntashirakon/setedit/utils/SettingsUtils.java:
```diff
@@ -30,21 +30,21 @@ public static ActionResult update(@NonNull Context context, @SettingsType String
                                       @NonNull String keyName) {
         if (Boolean.TRUE.equals(Shell.isAppGrantedRoot())) {
-            Shell.Result result = Shell.cmd("settings delete " + settingsType + " " + keyName).exec();
+            Shell.Result result = Shell.cmd(commandBuilder(settingsType) +" delete " + settingsType + " " + keyName).exec();
```
This won't work for LOS settings as they will return `null`. You need to find a way to avoid this.

MuntashirAkon commented on app/src/main/java/io/github/muntashirakon/setedit/utils/SettingsUtils.java:
```diff
@@ -59,23 +59,23 @@ private static ActionResult updateInternal(@NonNull Context context, @SettingsTy
                                                @ActionResult.ActionType int actionType) {
         if (Boolean.TRUE.equals(Shell.isAppGrantedRoot())) {
-            Shell.Result result = Shell.cmd("settings put " + settingsType + " " + keyName + " \"" + newValue + "\"").exec();
+            Shell.Result result = Shell.cmd(commandBuilder(settingsType) + " put " + settingsType + " " + keyName + " \"" + newValue + "\"").exec();
```
Same goes here (e.g., `null` for LOS).

MuntashirAkon reviewed (changes requested):
Looks mostly okay, except the 4 issues that needs to be addressed. Thanks.

Tharow committed:
Remove spaces from AndroidManifest.xml
Found and implemented LOS settings command in SettingsUtils.java
Further edits made to inform user of root requirements for motorola table.

Signed-off-by: Theray Tharow <tharowt@tharow.net>

Tharow commented:
After personal testing, prompt on attempted edit of Motorola table without root, was added

Tharow committed:
Couldn't get ask for root popup to work without crashing
So canEdit is false for the motorola table unless the the user somehow has access. in the similar way to the AndroidPropertiesRecyclerAdapter.java

Signed-off-by: Theray Tharow <tharowt@tharow.net>

Tharow commented:
Prompt removed as it always causes a crashed after a permission fail.
canEdit is used to check whether or not it can be edited.
